### PR TITLE
Upgrade to Wagtail 3 initial

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

This PR upgrades wagtail to version 3.0

Related PRs/issues: #9674

# Checklist

**Tests**
- [x] Is the code I'm adding covered by tests?

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~ no
- [ ] ~~Did I squash my migration?~~ No, Just 2 new migrations files added for the upgrade
- [ ] ~~Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~ Deploy is scheduled

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

File: requirements.in
Comment (kevinhowbrook) [inline requirements.in:39]: **todo**

This will need testing and updating on the actual fork, I've forked it again for efficiency here

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/relations.py
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/research_hub/relations.py:24]: I am not sure if that is possible, but since we now use the `FieldPanel` can we change the import to `wagtail.admin import panels` and use `panels.FieldPanel`?

**Review by tbrlpld — COMMENTED** [review]
Looking good. Left only some tiny questions.

File: network-api/networkapi/urls.py
Comment (kevinhowbrook) [reply network-api/networkapi/urls.py:30]: :+1: my find and replace wasn't that sophisticated :rofl:

File: network-api/networkapi/tests/tests.py
Comment (tbrlpld) [inline network-api/networkapi/tests/tests.py:1]: Is this file created by a merge conflict somehow?

File: requirements.in
Comment (tbrlpld) [reply requirements.in:38]: I left one question on that footnotes PR

**Review by danielfmiranda — APPROVED** [review]
Hi @kevinhowbrook thanks for taking this work on! Very exciting to be working with a newer version of wagtail and also interesting to see how in this version almost all panels are replaced by `FieldPanel`.

After giving the code changes a quick look through, and also poking around in my local instance, everything is looking good to me! 

Approved 👍

File: network-api/networkapi/tests/tests.py
Comment (kevinhowbrook) [reply network-api/networkapi/tests/tests.py]: I've actually no idea where this came from - it's certainly not from me but yea I imagine it came in from a rebase on main. This file isn't currently on main so I've removed it, thanks!

File: requirements.in
Comment (kevinhowbrook) [reply requirements.in:38]: (for clarity) 

> Is the translatability maybe something that is worth contributing back to the original package?

@tbrlpld - I'm unsure, given it wasn't originally contributed I don't know if I can answer that. But what I can say is that all we need is the wagtail requirement altering for our wagtail upgrade. I would suggest that yes, this is something we should merge into footnotes - but it will need more work as I think the migrations are out of sync with the fork and it's likely to hold up the upgrade for a bit.

Comment (kevinhowbrook) [reply requirements.in]: re-targeted in 2228dc9fab58fa956b2c4c553a38168aa50cd95b

**Comment by kevinhowbrook** [comment]
**NOTE:** We are holding off merging this for the moment until internal comms have been made about the upgrade and what to expect for editors.

**Comment by kevinhowbrook** [comment]
Some final testing before merge

- [ ] Tests pass with stage data
- [ ] Migrations fine with stage data
- [ ] Tests pass with prod data
- [ ] Migrations fine with prod data
- [ ] Tests and migrations all ok with local new db
- [ ] new build can be set up

**Pull request merged**
